src/dataframe/dataframe.jl

```diff
@@ -406,14 +409,13 @@ end
 function Base.setindex!(df::DataFrame,
                 val::Any,
                 col_inds::AbstractVector{Bool})
-    setindex!(df, val, find(col_inds))
+    df[find(col_inds)] = val
```

Comment:
The return value of `setindex!` is the dataframe; the return value of `x[] = y` is the value assigned.